src/http/matcher/mod.rs

```diff
@@ -65,9 +74,49 @@ pub enum HttpMatcherKind {
     ///
     /// [`SocketAddr`]: std::net::SocketAddr
     Socket(SocketMatcher),
+    /// A custom matcher that implements [`crate::service::Matcher`].
+    Custom(Arc<dyn crate::service::Matcher<State, Request<Body>>>),
+}
+
+impl<State, Body> Clone for HttpMatcherKind<State, Body> {
+    fn clone(&self) -> Self {
+        match self {
+            Self::All(inner) => Self::All(inner.clone()),
+            Self::Method(inner) => Self::Method(*inner),
+            Self::Path(inner) => Self::Path(inner.clone()),
+            Self::Domain(inner) => Self::Domain(inner.clone()),
+            Self::Version(inner) => Self::Version(*inner),
+            Self::Any(inner) => Self::Any(inner.clone()),
+            Self::Uri(inner) => Self::Uri(inner.clone()),
+            Self::Header(inner) => Self::Header(inner.clone()),
+            Self::Socket(inner) => Self::Socket(inner.clone()),
+            Self::Custom(inner) => Self::Custom(inner.clone()),
+        }
+    }
+}
+
+impl<State, Body> fmt::Debug for HttpMatcherKind<State, Body>
+where
+    State: fmt::Debug,
```

Comment:
Tell me if the compiler says otherwise, but pretty certain you do no need these trait bounds for `State` / `Body` here, neither are they desired.

Reply:
You are correct, they could be removed once I added the manual `Debug` impl.